apps/vpnmanager/src/components/Statistics/Statistics.tsx

```diff
@@ -0,0 +1,313 @@
+import React, { useEffect, useRef, useState } from "react";
+import {
+  Box,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Paper,
+  TextField,
+  Grid,
+  Button,
+  TablePagination,
+} from "@mui/material";
+import { Section } from "@commons-ui/core";
+import { useRouter } from "next/router";
+
+import formatDateTime, { formatDate } from "@/vpnmanager/utils/formatDate";
+import { fetchJson } from "@/vpnmanager/utils";
+import { Link } from "@commons-ui/next";
+
+export interface Data {
+  ID: number;
+  userId: string;
+  usage: number;
+  date: string;
+  cumulativeData: number;
+  email: string;
+  createdAt: string;
+}
+
+interface Props {
+  data: Data[];
+}
+
+const Statistics: React.FC<Props> = ({ data: result }) => {
+  const router = useRouter();
+  const yesterday = new Date();
+  yesterday.setDate(yesterday.getDate() - 1);
+  const [filters, setFilters] = useState({
+    ID: router.query.ID || "",
+    userId: router.query.userId || "",
+    email: router.query.email || "",
+    "date.start": router.query["date.start"] || "",
+    "date.end": router.query["date.end"] || "",
+    date: router.query.date || formatDate(yesterday),
+    orderBy: "date DESC",
+  });
+  const [page, setPage] = useState(0);
+  const [rowsPerPage, setRowsPerPage] = useState(10);
+  const [data, setData] = useState(result);
+  const paginatedData = data.slice(
+    page * rowsPerPage,
+    page * rowsPerPage + rowsPerPage,
+  );
+  useEffect(() => {
+    (async () => {
+      try {
+        const params: any = {};
+        Object.keys(filters).forEach((key) => {
+          if (filters[key as keyof typeof filters]) {
+            params[key] = filters[key as keyof typeof filters];
+          }
+        });
+        const res = await fetchJson.get("/api/statistics", {
+          params: params as Record<string, string>,
```

Comment:
Overriding the type here is a little tricky, let's satisfy it instead. let's declare params as type `Record<keyof typeof filters, string>` then work to make the iteration above compliant as well using `Object.fromEntries()`:

```
...
const params: Partial<Record<keyof typeof filters, string>> = Object.fromEntries(
          Object.entries(filters).filter(([_, value]) => Boolean(value)),
        );

        const res = await fetchJson.get("/api/statistics", {
          params: params,
        });
...
```